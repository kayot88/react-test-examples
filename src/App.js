import React from 'react';
import './App.css';
import NavBar from './components/navBar';
import LoginForm from './components/loginForm';

function App() {
  return (
    <div className="App">
      <NavBar />
      <LoginForm />
    </div>
  );
}

export default App;
