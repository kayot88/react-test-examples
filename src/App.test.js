import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

xtest('demo testing', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/welcome/i);
  expect(linkElement).toBeInTheDocument();
});
