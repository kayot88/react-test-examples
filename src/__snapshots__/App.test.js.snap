// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapShot testing 1`] = `
<div
  className="App"
>
  <div
    className="navbar"
  >
    <a
      data-testid="company-logo"
      href="/"
    >
      <img
        alt="Company Logo"
        src="/logo192.png"
      />
    </a>
    <ul>
      <li>
        <a
          href="/"
        >
           Home 
        </a>
      </li>
      <li>
        <a
          href="/about"
        >
           About 
        </a>
      </li>
      <li>
        <a
          href="/contact"
        >
           Contact 
        </a>
      </li>
      <li>
        <a
          href="/search"
        >
           Search 
        </a>
      </li>
    </ul>
  </div>
  <div>
    <h1>
      Anywhere in your app!
    </h1>
    <form
      onSubmit={[Function]}
    >
      <label>
        Name:
        <input
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter your name here"
          type="text"
          value=""
        />
      </label>
      <p
        style={
          Object {
            "color": "red",
          }
        }
      />
      <label>
        Email:
        <input
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Your Email Address"
          type="email"
          value=""
        />
      </label>
      <p
        style={
          Object {
            "color": "red",
          }
        }
      />
      <label>
        Password:
        <input
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="password here"
          type="password"
          value=""
        />
      </label>
      <p
        style={
          Object {
            "color": "red",
          }
        }
      />
      <button
        disabled={false}
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
